packages/components/message/src/message-method.ts

```diff
@@ -6,116 +6,121 @@ import { messageConfig } from '@element-plus/components/config-provider/src/conf
-      (item) =>
-        `${item.vm.props?.message ?? ''}` ===
-        `${(options as any).message ?? ''}`
-    )
-    if (tempVm) {
-      tempVm.vm.component!.props.repeatNum += 1
-      tempVm.vm.component!.props.type = options?.type
-      return {
-        close: () =>
-          ((
-            vm.component!.proxy as ComponentPublicInstance<{ visible: boolean }>
-          ).visible = false),
+    if (
+      !isVNode(options) &&
+      typeof options === 'object' &&
+      options.grouping &&
+      !isVNode(options.message) &&
+      instances.length
+    ) {
+      const tempVm: any = instances.find(
+        (item) =>
+          `${item.vm.props?.message ?? ''}` ===
+          `${(options as any).message ?? ''}`
+      )
+      if (tempVm) {
+        tempVm.vm.component!.props.repeatNum += 1
+        tempVm.vm.component!.props.type = options?.type
+        return {
+          close: () =>
+            ((
+              vm.component!.proxy as ComponentPublicInstance<{
+                visible: boolean
+              }>
+            ).visible = false),
+        }
       }
     }
-  }
 
-  if (typeof options === 'string' || isVNode(options)) {
-    options = { message: options }
-  }
+    if (typeof options === 'string' || isVNode(options)) {
+      options = { message: options }
+    }
 
-  let verticalOffset = options.offset || 20
-  instances.forEach(({ vm }) => {
-    verticalOffset += (vm.el?.offsetHeight || 0) + 16
-  })
-  verticalOffset += 16
-
-  const { nextZIndex } = useZIndex()
-
-  const id = `message_${seed++}`
-  const userOnClose = options.onClose
-  const props: Partial<MessageProps> = {
-    zIndex: nextZIndex(),
-    offset: verticalOffset,
-    ...options,
-    id,
-    onClose: () => {
-      close(id, userOnClose)
-    },
-  }
+    let verticalOffset = options.offset || 20
+    instances.forEach(({ vm }) => {
+      verticalOffset += (vm.el?.offsetHeight || 0) + 16
+    })
+    verticalOffset += 16
 
-  let appendTo: HTMLElement | null = document.body
-  if (options.appendTo instanceof HTMLElement) {
-    appendTo = options.appendTo
-  } else if (typeof options.appendTo === 'string') {
-    appendTo = document.querySelector(options.appendTo)
-  }
-  // should fallback to default value with a warning
-  if (!(appendTo instanceof HTMLElement)) {
-    debugWarn(
-      'ElMessage',
-      'the appendTo option is not an HTMLElement. Falling back to document.body.'
-    )
-    appendTo = document.body
-  }
+    const { nextZIndex } = useZIndex()
 
-  const container = document.createElement('div')
+    const id = `message_${seed++}`
+    const userOnClose = options.onClose
+    const props: Partial<MessageProps> = {
+      zIndex: nextZIndex(),
+      offset: verticalOffset,
+      ...options,
+      id,
+      onClose: () => {
+        close(id, userOnClose)
+      },
+    }
+
+    let appendTo: HTMLElement | null = document.body
+    if (options.appendTo instanceof HTMLElement) {
```

Comment:
should we have an util fn to check Element like `isHtmlElement()`?

Reply:
Yes we do, but that was in `components/images` we should extract that out.